src/query/phrase_query/phrase_scorer.rs

```diff
@@ -104,7 +104,8 @@ pub(crate) fn intersection_count(left: &[u32], right: &[u32]) -> usize {
 /// resulting array in left.
 ///
 /// Returns the length of the intersection
-fn intersection(left: &mut [u32], right: &[u32]) -> usize {
+#[inline]
+fn intersection(left: &mut Vec<u32>, right: &[u32]) -> usize {
```

Comment:
Does this need to return `count` if `left` is truncated anyway so that the information is "returned" via `left.len()`?

Reply:
No, but it's clear for the function caller

Reply:
What is the motivation for this change?

Reply:
It's simpler to use, you get the results in left, instead of `left[..returned size]`

Reply:
agreed. But can we remove the returned value?

Reply:
Yes, I'll create a benchmark for it

Reply:
the search benchmark phrase tag + count is ok too.

Reply:
Performance looks good
Type of Query: Phrase
tantivy-0.18 | tantivy-0.19 | tantivy-phrase
-- | -- | --
1,114 μs | 1,114 μs | 1,115 μs


TOP10_COUNT seems to have regressed a little though on current main (TOP10 and COUNT are fine)
Type of Query: All
Query | tantivy-0.18 | tantivy-0.19 | tantivy-phrase
-- | -- | -- | --
AVERAGE | 2,561 μs | 2,599 μs | 2,822 μs
the | 11,003  μs4,168,066 docs | 16,523  μs+50.2 %4,168,066 docs | 19,449  μs+76.8 %4,168,066 docs

Reply:
can you share results.json?

Reply:
[results.zip](https://github.com/quickwit-oss/tantivy/files/11369965/results.zip)

Reply:
> agreed. But currently the change is halfway, can we remove the returned value?

We need to return the count, in case left is not updated

Reply:
> [results.zip](https://github.com/quickwit-oss/tantivy/files/11369965/results.zip)

thx

Reply:
the results looks good!

Reply:
> We need to return the count, in case left is not updated

I don't understand. 
`left.len()` is not always the same as the result of the function?

Reply:
sorry I thought was about `intersection_count_with_slop`
returned count is never used for `intersection` now, I removed it